pymc/gp/cov.py

```diff
@@ -548,7 +568,48 @@ def full(self, X, Xs=None):
         return pt.exp(-0.5 * r)
 
 
-class ExpQuad(Stationary):
+class GeneralizedPeriodic(Stationary):
+    r"""
+    The Generalized Periodic kernel.
+
+    Can be used to wrap any isotropic stationary kernel to transform it
+    into a periodic version. The derivation can be achieved by mapping the
+    original inputs through the transformation u = (cos(x), sin(x)).
+    """
+
+    def __init__(self, base_kernel: IsotropicStationary, period):
+        self.base_kernel = base_kernel
+        self.period = period
+
+    @property
+    def input_dim(self):
+        return self.base_kernel.input_dim
+
+    @property
+    def active_dims(self):
+        return self.base_kernel.active_dims
+
+    @property
+    def ls(self):
+        return self.base_kernel.ls
+
+    def full(self, X, Xs=None):
+        X, Xs = self._slice(X, Xs)
+        if Xs is None:
+            Xs = X
+        f1 = X.dimshuffle(0, "x", 1)
+        f2 = Xs.dimshuffle("x", 0, 1)
```

Comment:
I really dislike `dimshuffle`! Could we switch to numpy syntax here?

```suggestion
        f1 = X[..., :, None, :]
        f2 = Xs[..., None, :, :]
```

you can also use `numpy.newaxis` if you find that more readable

Reply:
There's also `pytensor.tensor.expand_dims`

Reply:
Yes absolutely, I had just copied and pasted what was currently in `Periodic`